spec/coordinates_spec.rb

```diff
@@ -0,0 +1,30 @@
+require "spec_helper"
+require_relative "../lib/coordinates"
+
+RSpec.describe Coordinates do
+  describe "valid?" do
+    it "returns false when row is not valid" do
```

Comment:
This description is kinda like using the word in the definition of the word. What are the characteristics that make the row invalid?

E.g. since the grid is visual, a cell that's beyond the top/right/bottom/left is invalid. What do you think about making specs specifically to test those edge cases?

Reply:
Good call out. Done in 4035b4a8b4432bfb9dd86248e85b06289a131d18. Do you have a recommendation to test a "row before the first" is invalid. What's before "A"? hehe

Reply:
Yeah, before first row is weird. I used `@2`, because `@` is before `A` in ASCII, but then the test no longer made sense as "before the first row" and I changed it to being about malformed data. That's similar to the spec you have with `a43`

Reply:
hehe I like that you used `@`. I'm going to omit "before first row" then. What do you think about the edge cases we're testing now?

Reply:
They look good!